### PR TITLE
Add new target for checking modified zuul files

### PR DESCRIPTION
**Pull request by adrianfusco**
- Add new target for checking modified zuul files (Moved from https://github.com/openshift/release/pull/42730). It will re-generate the `zuul.d/molecule.yaml` and `zuul.d/projects.yaml` files, if they're not in commit phase or if they have been modified manually then the target will fail.
- Add check_zuul_files target in pre-comit-config file

If it's passing:

```
[2023-08-29 16:16:04] {afuscoar@afuscoar} (~/.Redhat/Projects/github/ci-framework) (feature/target-zuul-jobs-check)$ -> pre-commit run --all-files; 
make-check-zuul-files....................................................Passed
...
```
If it's not:

```
[2023-08-29 16:25:51] {afuscoar@afuscoar} (~/.Redhat/Projects/github/ci-framework) (feature/target-zuul-jobs-check)$ -> pre-commit run --all-files; 
make-check-zuul-files....................................................Failed
- hook id: make-check-zuul-files
- files were modified by this hook
...
```
- [x] Added description in new target

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -88,6 +88,21 @@ pre_commit_nodeps: ## Run pre-commit tests without installing dependencies
 		pre-commit run --all-files; \
 	fi
 
+.PHONY: check_zuul_files
```
Comment (cjeanner) [inline Makefile:91]: I love the idea - maybe we should nudge that in a script though, to make things a bit more readable?

Comment (adrianfusco) [reply Makefile:91]: That's right. Change done.

File: scripts/check_zuul_files.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+files=("zuul.d/molecule.yaml" "zuul.d/projects.yaml")
+files_with_diff=()
+for file in "${files[@]}"; do
+    if ! git diff --exit-code --quiet -- "$file" > /dev/null 2>&1; then
+        files_with_diff+=("$file")
+        echo "Changes detected in $file"
```
Comment (Sandeepyadav93) [inline scripts/check_zuul_files.sh:22]: May add some more details in the print


        echo "Manual changes are not allowed in this file."
        echo "This file is autogenerated using 'make role_molecule'."

**Review by Sandeepyadav93 — COMMENTED** [review]
This looks great, minor inline comment.

Tested locally - works great.

File: scripts/check_zuul_files.sh (hunk 2 above)
Comment (adrianfusco) [reply scripts/check_zuul_files.sh:23]: Added. Thx Sandeep.

File: scripts/check_zuul_files.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    fi
+done
+if [ ${#files_with_diff[@]} -gt 0 ]; then
```
[the remark this answers is not in the document]
Comment (adrianfusco) [reply scripts/check_zuul_files.sh]: Added quotes in variable. Thx Sandeep.

**Review by cjeanner — APPROVED** [review]
/approve

Thanks for that nice addition!

**Pull request merged**